smokescreen.go

```diff
@@ -217,17 +214,42 @@ func main() {
 		}
 	}
 
-	server := graceful.NewServer(&http.Server{
+	server := http.Server{
 		Handler: handler,
-	})
+	}
+	timeout := time.Second * time.Duration(60)
+
+	runServer(&server, listener, timeout)
+}
+
+func runServer(server *http.Server, listener net.Listener, timeout time.Duration) {
+	// Runs the server and shuts it down when it receives a signal.
+	//
+	// Why aren't we using goji's graceful shutdown library? Great question!
+	//
+	// There are several things we might want to do when shutting down gracefully:
+	// 1. close the listening socket (so that we don't accept *new* connections)
+	// 2. close *existing* keepalive connections once they become idle
+	//
+	// It is impossible to close existing keepalive connections, because goproxy
+	// hijacks the socket. So all we can do is close the listening socket when
```

Comment:
I'd add `because goproxy hijacks the socket and doesn't tell us when they become idle`